src/client/components/ProductsList.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+class ProductsList extends Component {
+  state = {
+    limit: 3
+  };
+
+  showMore = () => {
+    this.setState(({ limit }) => ({ limit: limit + 3 }));
+  };
+
+  render() {
+    const { products, isArchive } = this.props;
+    const { limit } = this.state;
+    return (
+      <Fragment>
+        <ul className="products-list">
+          {products.slice(0, limit).map(item => (
+            <li
```

Comment:
Please extract that code to separate stateless component (e.g ProductsListItem), or at least separate function, which will return list item.